mp/src/game/server/cbase.cpp

```diff
@@ -1305,18 +1311,44 @@ void CEventQueueEvent::FromPrioritizedEvent( const EventQueuePrioritizedEvent_t
 	m_iCaller = pe->m_pCaller ? pe->m_pCaller->entindex() : 0;
 	m_szEntTarget = pe->m_pEntTarget ? pe->m_pEntTarget->GetEntityName() : NULL_STRING;
 	m_VariantValue = pe->m_VariantValue;
+
+	if (pAbstractedEntity)
+	{
+		m_bAbstractTarget = (pe->m_pEntTarget == pAbstractedEntity);
+		m_bAbstractActivator = (pe->m_pActivator == pAbstractedEntity);
+		m_bAbstractCaller = (pe->m_pCaller == pAbstractedEntity);
+	}
+	else
+	{
+		m_bAbstractTarget = false;
+		m_bAbstractActivator = false;
+		m_bAbstractCaller = false;
+	}
 }
 
-void CEventQueueEvent::ToPrioritizedEvent(EventQueuePrioritizedEvent_t* pe) const
+void CEventQueueEvent::ToPrioritizedEvent(EventQueuePrioritizedEvent_t* pe, CBaseEntity *pAbstractedEntity ) const
 {
 	// Convert from tick offset to absolute tick count
 	pe->m_iFireTick = m_iFireDelayTicks + gpGlobals->tickcount;
 	pe->m_iOutputID = m_iOutputID;
 	pe->m_iTarget = m_iTarget;
 	pe->m_iTargetInput = m_iTargetInput;
-	pe->m_pActivator = GetEntityFromName( m_szActivator );
-	pe->m_pCaller = UTIL_EntityByIndex( m_iCaller );
-	pe->m_pEntTarget = GetEntityFromName( m_szEntTarget );;
+
+	if (pAbstractedEntity && m_bAbstractActivator)
+		pe->m_pActivator = pAbstractedEntity;
+	else
+		pe->m_pActivator = GetEntityFromName( m_szActivator );
+
+	if (pAbstractedEntity && m_bAbstractCaller)
+		pe->m_pCaller = pAbstractedEntity;
+	else
+		pe->m_pCaller = UTIL_EntityByIndex( m_iCaller );
+
+	if (pAbstractedEntity && m_bAbstractTarget)
+		pe->m_pEntTarget = pAbstractedEntity;
+	else
+		pe->m_pEntTarget = GetEntityFromName( m_szEntTarget );
```

Comment:
Small style thing but can you use curly braces for these branches? Old momentum code notwithstanding, it's good to wrap different branches with curlies, even if one line, and to reserve the one-liners for either a single logic branch:
```
if (bFoo)
    bar();

 . . . 
```

Or for early returns:
```
if (bEarlyFoo)
    return;
```